python/ray/data/datasource/file_based_datasource.py

```diff
@@ -792,3 +770,31 @@ def _resolve_kwargs(
         kwarg_overrides = kwargs_fn()
         kwargs.update(kwarg_overrides)
     return kwargs
+
+
+Uri = TypeVar("Uri")
+Meta = TypeVar("Meta")
+
+
+def _fetch_metadata_parallel(
+    uris: List[Uri],
+    fetch_func: Callable[[List[Uri]], List[Meta]],
```

Comment:
The `_fetch_metadata_serialization_wrapper` look taking _SerializedPiece typed arg.

Reply:
Yep, and the provided `uris` arg is a `List[_SerializedPiece]`, so the `Uri` type is consistent across these two args, which is all that we're trying to express here (`fetch_func` should take the same type as `uris` as an argument). I couldn't think of a better type name than `Uri`, since both file paths and Parquet fragments can be thought of as an identifier for data to be read, so `Uri` seemed like an ok choice.